Error while checking build/O2/fullCI for 38c34dc5d89ba50b501857a256fca5f36aec7970 at 2023-02-15 11:09:
```
## sw/BUILD/o2checkcode-latest/log
/sw/SOURCES/O2/10769-slc8_x86-64/0/GPU/Common/test/testGPUsortCUDA.cu:22:10: error: 'boost/test/unit_test.hpp' file not found [clang-diagnostic-error]
/sw/SOURCES/O2/10769-slc8_x86-64/0/GPU/GPUTracking/TRDTracking/GPUTRDTracker.cxx:37:10: error: 'omp.h' file not found [clang-diagnostic-error]
/sw/SOURCES/O2/10769-slc8_x86-64/0/GPU/GPUTracking/Base/GPUReconstruction.cxx:37:10: error: 'omp.h' file not found [clang-diagnostic-error]
/sw/SOURCES/O2/10769-slc8_x86-64/0/GPU/GPUTracking/Base/GPUReconstructionCPU.cxx:45:10: error: 'omp.h' file not found [clang-diagnostic-error]
/sw/SOURCES/O2/10769-slc8_x86-64/0/GPU/GPUTracking/display/GPUDisplay.cxx:36:10: error: 'omp.h' file not found [clang-diagnostic-error]
/sw/SOURCES/O2/10769-slc8_x86-64/0/GPU/GPUTracking/Base/cuda/GPUReconstructionCUDAGenRTC.cu:16:10: error: 'omp.h' file not found [clang-diagnostic-error]
/sw/SOURCES/O2/10769-slc8_x86-64/0/GPU/GPUbenchmark/cuda/../Shared/Utils.h:26:10: error: 'boost/program_options.hpp' file not found [clang-diagnostic-error]
/sw/SOURCES/O2/10769-slc8_x86-64/0/GPU/GPUbenchmark/cuda/../Shared/Utils.h:26:10: error: 'boost/program_options.hpp' file not found [clang-diagnostic-error]
/sw/SOURCES/O2/10769-slc8_x86-64/0/Framework/Logger/include/Framework/Logger.h:14:10: error: 'fairlogger/Logger.h' file not found [clang-diagnostic-error]
/usr/local/cuda-12.0/include/cuda/std/detail/libcxx/include/type_traits:520:12: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-12.0/include/cuda/std/detail/libcxx/include/type_traits:1059:69: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-12.0/include/cuda/std/detail/libcxx/include/type_traits:1175:5: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-12.0/include/cuda/std/detail/libcxx/include/type_traits:1852:16: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-12.0/include/cuda/std/detail/libcxx/include/type_traits:2276:18: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-12.0/include/cuda/std/detail/libcxx/include/type_traits:4336:16: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-12.0/include/sm_20_atomic_functions.h:103:39: error: redefinition of 'atomicAdd' [clang-diagnostic-error]
/sw/SOURCES/O2/10769-slc8_x86-64/0/Framework/Logger/include/Framework/Logger.h:14:10: error: 'fairlogger/Logger.h' file not found [clang-diagnostic-error]
/usr/local/cuda-12.0/include/cuda/std/detail/libcxx/include/type_traits:520:12: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-12.0/include/cuda/std/detail/libcxx/include/type_traits:1059:69: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-12.0/include/cuda/std/detail/libcxx/include/type_traits:1175:5: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-12.0/include/cuda/std/detail/libcxx/include/type_traits:1852:16: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-12.0/include/cuda/std/detail/libcxx/include/type_traits:2276:18: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-12.0/include/cuda/std/detail/libcxx/include/type_traits:4336:16: error: CUDA device code does not support variadic functions [clang-diagnostic-error]
/usr/local/cuda-12.0/include/sm_20_atomic_functions.h:103:39: error: redefinition of 'atomicAdd' [clang-diagnostic-error]
/sw/SOURCES/O2/10769-slc8_x86-64/0/GPU/GPUbenchmark/hip/../Shared/Utils.h:146:34: error: use of undeclared identifier 'int4'; did you mean 'int'? [clang-diagnostic-error]
/sw/SOURCES/O2/10769-slc8_x86-64/0/GPU/GPUbenchmark/hip/benchmark.hip.cxx:199:35: error: use of undeclared identifier 'int4'; did you mean 'int'? [clang-diagnostic-error]
/sw/SOURCES/O2/10769-slc8_x86-64/0/GPU/GPUbenchmark/hip/../Shared/Utils.h:146:34: error: use of undeclared identifier 'int4'; did you mean 'int'? [clang-diagnostic-error]
/sw/SOURCES/O2/10769-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:64:8: error: unknown type name '__host__' [clang-diagnostic-error]
/sw/SOURCES/O2/10769-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:64:27: error: expected ';' after top level declarator [clang-diagnostic-error]
/sw/SOURCES/O2/10769-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:64:33: error: overloaded 'operator+=' must have at least one parameter of class or enumeration type [clang-diagnostic-error]
/sw/SOURCES/O2/10769-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:64:44: error: unknown type name 'int4'; did you mean 'int'? [clang-diagnostic-error]
/sw/SOURCES/O2/10769-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:64:53: error: unknown type name 'int4'; did you mean 'int'? [clang-diagnostic-error]
/sw/SOURCES/O2/10769-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:66:4: error: member reference base type 'int' is not a structure or union [clang-diagnostic-error]
/sw/SOURCES/O2/10769-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:66:11: error: member reference base type 'int' is not a structure or union [clang-diagnostic-error]
/sw/SOURCES/O2/10769-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:67:4: error: member reference base type 'int' is not a structure or union [clang-diagnostic-error]
/sw/SOURCES/O2/10769-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:67:11: error: member reference base type 'int' is not a structure or union [clang-diagnostic-error]
/sw/SOURCES/O2/10769-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:68:4: error: member reference base type 'int' is not a structure or union [clang-diagnostic-error]
/sw/SOURCES/O2/10769-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:68:11: error: member reference base type 'int' is not a structure or union [clang-diagnostic-error]
/sw/SOURCES/O2/10769-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:69:4: error: member reference base type 'int' is not a structure or union [clang-diagnostic-error]
/sw/SOURCES/O2/10769-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:69:11: error: member reference base type 'int' is not a structure or union [clang-diagnostic-error]
/sw/SOURCES/O2/10769-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:85:1: error: unknown type name '__global__' [clang-diagnostic-error]
/sw/SOURCES/O2/10769-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:90:19: error: use of undeclared identifier 'blockIdx' [clang-diagnostic-error]
/sw/SOURCES/O2/10769-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:90:32: error: use of undeclared identifier 'blockDim' [clang-diagnostic-error]
/sw/SOURCES/O2/10769-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:90:45: error: use of undeclared identifier 'threadIdx' [clang-diagnostic-error]
/sw/SOURCES/O2/10769-slc8_x86-64/0/GPU/GPUbenchmark/hip/Kernels.hip.cxx:90:78: error: use of undeclared identifier 'blockDim' [clang-diagnostic-error]
/sw/slc8_x86-64/GCC-Toolchain/v12.2.0-alice1-2/lib/gcc/x86_64-unknown-linux-gnu/12.2.0/../../../../include/c++/12.2.0/bits/shared_ptr_base.h:196:22: error: expected expression [clang-diagnostic-error]
/sw/slc8_x86-64/GCC-Toolchain/v12.2.0-alice1-2/lib/gcc/x86_64-unknown-linux-gnu/12.2.0/../../../../include/c++/12.2.0/bits/shared_ptr_base.h:196:22: error: type name does not allow function specifier to be specified [clang-diagnostic-error]
/sw/slc8_x86-64/GCC-Toolchain/v12.2.0-alice1-2/lib/gcc/x86_64-unknown-linux-gnu/12.2.0/../../../../include/c++/12.2.0/bits/shared_ptr_base.h:196:22: error: use of undeclared identifier 'noinline'; did you mean 'inline'? [clang-diagnostic-error]
/sw/SOURCES/O2/10769-slc8_x86-64/0/Detectors/EMCAL/calibration/include/EMCALCalibration/EMCALCalibExtractor.h:36:10: error: 'omp.h' file not found [clang-diagnostic-error]
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/10769/38c34dc5d89ba50b501857a256fca5f36aec7970/build_O2_fullCI/pretty.html).

--- a/Detectors/ITSMFT/MFT/condition/CMakeLists.txt
+++ b/Detectors/ITSMFT/MFT/condition/CMakeLists.txt
@@ -10,15 +10,15 @@
 
 o2_add_library(MFTCondition
                SOURCES src/MFTDCSProcessor.cxx
-                       src/DCSNameResolver.cxx                       
-		       src/DCSConfigInfo.cxx
-		       src/DCSConfigReader.cxx
+                       src/DCSNameResolver.cxx
+                       src/DCSConfigInfo.cxx
+                       src/DCSConfigReader.cxx
                        src/DCSConfigUtils.cxx
                PUBLIC_LINK_LIBRARIES O2::CCDB
+                                     O2::DataFormatsMFT
                                      O2::DetectorsCalibration
                                      O2::DetectorsDCS
                                      Microsoft.GSL::GSL)
-
 o2_target_root_dictionary(MFTCondition
                           HEADERS include/MFTCondition/MFTDCSProcessor.h
                                   include/MFTCondition/DCSNameResolver.h

--- a/Detectors/ITSMFT/MFT/condition/include/MFTCondition/DCSConfigInfo.h
+++ b/Detectors/ITSMFT/MFT/condition/include/MFTCondition/DCSConfigInfo.h
@@ -68,7 +68,7 @@ class DCSConfigInfo
  private:
   int mData;
   int mAdd;
-  int mType; // RU = 0, ALPIDE = 1
+  int mType; // RU = 0, ALPIDE = 1, UBB = 2, DeadMap = 3
   std::string mVersion;
 
   ClassDefNV(DCSConfigInfo, 1);

--- a/Detectors/ITSMFT/MFT/condition/include/MFTCondition/DCSConfigReader.h
+++ b/Detectors/ITSMFT/MFT/condition/include/MFTCondition/DCSConfigReader.h
@@ -13,7 +13,9 @@
 #define MFTDCSCONFIGPROCESSOR_H_
 
 #include "Rtypes.h"
+#include "DataFormatsITSMFT/NoiseMap.h"
 #include "MFTCondition/DCSConfigInfo.h"
+#include "MFTCondition/DCSConfigUtils.h"
 #include <gsl/span>
 #include <memory>
 
@@ -37,20 +39,19 @@ class DCSConfigReader
   void loadConfig(gsl::span<const char> configBuf); // load MFT config
   void clear();
 
-  std::vector<o2::mft::DCSConfigInfo>& getConfigInfo() { return mDCSConfig; }
+  const std::vector<o2::mft::DCSConfigInfo>& getConfigInfo() const { return mDCSConfig; }
+  const o2::itsmft::NoiseMap& getNoiseMap() const { return mNoiseMap; }
 
  private:
   void parseConfig();
 
   std::string mParams;
-
-  int mNumRow;
-  int mNumRU;
-  int mNumALPIDE;
-
   bool mVerbose = false;
 
   std::vector<o2::mft::DCSConfigInfo> mDCSConfig;
+  o2::itsmft::NoiseMap mNoiseMap{936};
+
+  DCSConfigUtils* mDCSUtils;
 
   ClassDefNV(DCSConfigReader, 1);
 };

--- a/Detectors/ITSMFT/MFT/condition/include/MFTCondition/DCSConfigUtils.h
+++ b/Detectors/ITSMFT/MFT/condition/include/MFTCondition/DCSConfigUtils.h
@@ -35,9 +35,6 @@ class DCSConfigUtils
   {
     init();
   }
-
-  ~DCSConfigUtils() = default;
-
   void init()
   {
     initDictionary();
@@ -47,34 +44,34 @@ class DCSConfigUtils
   {
   }
 
-  auto getAddress(const std::string name, const std::string type)
+  int getAddress(std::string name, std::string type)
   {
-    return static_cast<int>((mNameDictType[type])[name]);
+    return (mNameDict[type])[name];
   }
 
-  const std::string getName(int add, const std::string type)
+  std::string getName(int add, std::string type)
   {
-    return static_cast<const std::string>((mAddressDictType[type])[add]);
+    return (mAddressDict[type])[add];
   }
 
-  std::map<int, const std::string>& getAddressMap(const std::string type)
+  std::map<int, std::string> getAddressMap(std::string type)
   {
-    return mAddressDictType[type];
+    return mAddressDict[type];
   }
 
-  std::map<int, const std::string>& getAddressMap(int type)
+  std::map<int, std::string> getAddressMap(int type)
   {
-    return mAddressDictType[getTypeName(type)];
+    return mAddressDict[getTypeName(type)];
   }
 
-  std::map<const std::string, int>& getNameMap(const std::string type)
+  std::map<std::string, int> getNameMap(std::string type)
   {
-    return mNameDictType[type];
+    return mNameDict[type];
   }
 
-  std::map<const std::string, int>& getNameMap(int type)
+  std::map<std::string, int> getNameMap(int type)
   {
-    return mNameDictType[getTypeName(type)];
+    return mNameDict[getTypeName(type)];
   }
 
   const std::string getTypeName(int type)
@@ -93,7 +90,7 @@ class DCSConfigUtils
     }
   }
 
-  const int getType(const std::string type)
+  const int getType(std::string type)
   {
     if (type == "RU") {
       return 0;
@@ -109,171 +106,153 @@ class DCSConfigUtils
     }
   }
 
-  const std::string& getVersion()
+  std::string& getVersion()
   {
     return mVersion;
   }
 
-  auto& getVersionNameLineInCsv() const
+  int getVersionNameLineInCsv()
   {
-    return mVersionNameLine;
+    return mVerNameLine;
   }
-  auto& getRuConfigAddressLineInCsv() const
+  int getRUConfigAddressLineInCsv()
   {
-    return mRuConfigAddressLine;
+    return mRUConfAddLine;
   }
-  auto& getRuConfigValueLineInCsv() const
+  int getRUConfigValueLineInCsv()
   {
-    return mRuConfigValueLine;
+    return mRUConfValLine;
   }
-  auto& getAlpideConfigAddressLineInCsv() const
+  int getALPIDEConfigAddressLineInCsv()
   {
-    return mAlpideConfigAddressLine;
+    return mALPIDEConfAddLine;
   }
-  auto& getAlpideConfigValueLineInCsv() const
+  int getALPIDEConfigValueLineInCsv()
   {
-    return mAlpideConfigValueLine;
+    return mALPIDEConfValLine;
   }
-  auto& getUbbConfigNameLineInCsv() const
+  int getUBBConfigNameLineInCsv()
   {
-    return mUbbNameLine;
+    return mUBBNameLine;
   }
-  auto& getUbbConfigValueLineInCsv() const
+  int getUBBConfigValueLineInCsv()
   {
-    return mUbbValueLine;
+    return mUBBValLine;
   }
-  auto& getDeadMapLineInCsv() const
+  int getDeadMapLineInCsv()
   {
     return mDeadMapLine;
   }
 
  private:
-  const int mVersionNameLine = 0;
-  const int mAlpideConfigAddressLine = 1;
-  const int mAlpideConfigValueLine = 2;
-  const int mRuConfigAddressLine = 3;
-  const int mRuConfigValueLine = 4;
-  const int mUbbNameLine = 5;
-  const int mUbbValueLine = 6;
+  const int mVerNameLine = 0;
+  const int mALPIDEConfAddLine = 1;
+  const int mALPIDEConfValLine = 2;
+  const int mRUConfAddLine = 3;
+  const int mRUConfValLine = 4;
+  const int mUBBNameLine = 5;
+  const int mUBBValLine = 6;
   const int mDeadMapLine = 7;
 
-  const std::string mVersion;
+  std::string mVersion;
 
-  std::map<const std::string, std::map<const std::string, int>> mNameDictType;
-  std::map<const std::string, int> mNameDictRu;
-  std::map<const std::string, int> mNameDictAlpide;
-  std::map<const std::string, int> mNameDictUbb;
-  std::map<const std::string, int> mNameDictDeadMap;
+  std::map<std::string, std::map<std::string, int>> mNameDict;
+  std::map<std::string, int> mNameDictRU;
+  std::map<std::string, int> mNameDictALPIDE;
+  std::map<std::string, int> mNameDictUBB;
 
-  std::map<const std::string, std::map<int, const std::string>> mAddressDictType;
-  std::map<int, const std::string> mAddressDictRu;
-  std::map<int, const std::string> mAddressDictAlpide;
-  std::map<int, const std::string> mAddressDictUbb;
-  std::map<int, const std::string> mAddressDictDeadMap;
+  std::map<std::string, std::map<int, std::string>> mAddressDict;
+  std::map<int, std::string> mAddressDictRU;
+  std::map<int, std::string> mAddressDictALPIDE;
+  std::map<int, std::string> mAddressDictUBB;
 
   void initDictionary()
   {
-    mNameDictType.clear();
-    mNameDictRu.clear();
-    mNameDictAlpide.clear();
-    mNameDictUbb.clear();
-    mNameDictDeadMap.clear();
+    mNameDict.clear();
+    mNameDictRU.clear();
+    mNameDictALPIDE.clear();
+    mNameDictUBB.clear();
 
-    mAddressDictType.clear();
-    mAddressDictRu.clear();
-    mAddressDictAlpide.clear();
-    mAddressDictUbb.clear();
-    mAddressDictDeadMap.clear();
+    mAddressDict.clear();
+    mAddressDictRU.clear();
+    mAddressDictALPIDE.clear();
+    mAddressDictUBB.clear();
 
-    std::vector<std::pair<const std::string, uint>> pairRu;
-    pairRu.push_back(std::make_pair("MANCHESTER", 1046));
-    pairRu.push_back(std::make_pair("ENABLE", 4096));
-    pairRu.push_back(std::make_pair("TRIGGER_PERIOD", 4097));
-    pairRu.push_back(std::make_pair("PULSE_nTRIGGER", 4098));
-    pairRu.push_back(std::make_pair("TRIGGER_MIN_DISTANCE", 4099));
-    pairRu.push_back(std::make_pair("OPCODE_GATING", 4101));
-    pairRu.push_back(std::make_pair("TRIGGER_DELAY", 4102));
-    pairRu.push_back(std::make_pair("ENABLE_PACKER_0", 4103));
-    pairRu.push_back(std::make_pair("ENABLE_PACKER_1", 4104));
-    pairRu.push_back(std::make_pair("ENABLE_PACKER_2", 4105));
-    pairRu.push_back(std::make_pair("TRIG_SOURCE", 4106));
-    pairRu.push_back(std::make_pair("TIMEOUT_TO_START", 5376));
-    pairRu.push_back(std::make_pair("TIMEOUT_TO_STOP", 5377));
-    pairRu.push_back(std::make_pair("TIMEOUT_IN_IDLE", 5378));
-    pairRu.push_back(std::make_pair("GBT_LOAD_BALANCING", 5631));
+    std::vector<std::pair<std::string, uint>> pairRU;
+    pairRU.push_back(std::make_pair("MANCHESTER", 1046));
+    pairRU.push_back(std::make_pair("ENABLE", 4096));
+    pairRU.push_back(std::make_pair("TRIGGER_PERIOD", 4097));
+    pairRU.push_back(std::make_pair("PULSE_nTRIGGER", 4098));
+    pairRU.push_back(std::make_pair("TRIGGER_MIN_DISTANCE", 4099));
+    pairRU.push_back(std::make_pair("OPCODE_GATING", 4101));
+    pairRU.push_back(std::make_pair("TRIGGER_DELAY", 4102));
+    pairRU.push_back(std::make_pair("ENABLE_PACKER_0", 4103));
+    pairRU.push_back(std::make_pair("ENABLE_PACKER_1", 4104));
+    pairRU.push_back(std::make_pair("ENABLE_PACKER_2", 4105));
+    pairRU.push_back(std::make_pair("TRIG_SOURCE", 4106));
+    pairRU.push_back(std::make_pair("TIMEOUT_TO_START", 5376));
+    pairRU.push_back(std::make_pair("TIMEOUT_TO_STOP", 5377));
+    pairRU.push_back(std::make_pair("TIMEOUT_IN_IDLE", 5378));
+    pairRU.push_back(std::make_pair("GBT_LOAD_BALANCING", 5631));
 
-    std::vector<std::pair<const std::string, uint>> pairAlpide;
-    pairAlpide.push_back(std::make_pair("Mode_Control_Register", 1));
-    pairAlpide.push_back(std::make_pair("FROMU_Configration_Register_1", 4));
-    pairAlpide.push_back(std::make_pair("FROMU_Configration_Register_2", 5));
-    pairAlpide.push_back(std::make_pair("FROMU_Configration_Register_3", 6));
-    pairAlpide.push_back(std::make_pair("FROMU_Pulsing_Register_1", 7));
-    pairAlpide.push_back(std::make_pair("FROMU_Pulsing_Register_2", 8));
-    pairAlpide.push_back(std::make_pair("CMUandDMU_Configration_Register", 16));
-    pairAlpide.push_back(std::make_pair("DTU_Configration_Register", 20));
-    pairAlpide.push_back(std::make_pair("DTU_DACs_Register", 21));
-    pairAlpide.push_back(std::make_pair("DTU_Test_Register_1", 24));
-    pairAlpide.push_back(std::make_pair("DTU_Test_Register_2", 25));
-    pairAlpide.push_back(std::make_pair("DTU_Test_Register_3", 26));
-    pairAlpide.push_back(std::make_pair("VCASP", 1539));
-    pairAlpide.push_back(std::make_pair("VCLIP", 1544));
-    pairAlpide.push_back(std::make_pair("IBIAS", 1549));
-    pairAlpide.push_back(std::make_pair("VPULSEH", 255));
-    pairAlpide.push_back(std::make_pair("VPULSEL", 0));
-    pairAlpide.push_back(std::make_pair("VRESETD", 1538));
-    pairAlpide.push_back(std::make_pair("IDB", 1548));
-    pairAlpide.push_back(std::make_pair("AUTO_ROF__NOISE_MASK__MASK_LEV__MC_HIT__MC_ID", 65535));
+    std::vector<std::pair<std::string, uint>> pairALPIDE;
+    pairALPIDE.push_back(std::make_pair("Mode_Control_Register", 1));
+    pairALPIDE.push_back(std::make_pair("FROMU_Configration_Register_1", 4));
+    pairALPIDE.push_back(std::make_pair("FROMU_Configration_Register_2", 5));
+    pairALPIDE.push_back(std::make_pair("FROMU_Configration_Register_3", 6));
+    pairALPIDE.push_back(std::make_pair("FROMU_Pulsing_Register_1", 7));
+    pairALPIDE.push_back(std::make_pair("FROMU_Pulsing_Register_2", 8));
+    pairALPIDE.push_back(std::make_pair("CMUandDMU_Configration_Register", 16));
+    pairALPIDE.push_back(std::make_pair("DTU_Configration_Register", 20));
+    pairALPIDE.push_back(std::make_pair("DTU_DACs_Register", 21));
+    pairALPIDE.push_back(std::make_pair("DTU_Test_Register_1", 24));
+    pairALPIDE.push_back(std::make_pair("DTU_Test_Register_2", 25));
+    pairALPIDE.push_back(std::make_pair("DTU_Test_Register_3", 26));
+    pairALPIDE.push_back(std::make_pair("VCASP", 1539));
+    pairALPIDE.push_back(std::make_pair("VCLIP", 1544));
+    pairALPIDE.push_back(std::make_pair("IBIAS", 1549));
+    pairALPIDE.push_back(std::make_pair("VPULSEH", 255));
+    pairALPIDE.push_back(std::make_pair("VPULSEL", 0));
+    pairALPIDE.push_back(std::make_pair("VRESETD", 1538));
+    pairALPIDE.push_back(std::make_pair("IDB", 1548));
+    pairALPIDE.push_back(std::make_pair("AUTO_ROF__NOISE_MASK__MASK_LEV__MC_HIT__MC_ID", 65535));
 
-    std::vector<std::pair<const std::string, uint>> pairUbb;
-    int idUbb = 0;
+    std::vector<std::pair<std::string, uint>> pairUBB;
+    int idUBB = 0;
     for (int iH = 0; iH <= 1; ++iH) {
       for (int iD = 0; iD <= 4; ++iD) {
         for (int iF = 0; iF <= 1; ++iF) {
           for (int iZ = 0; iZ <= 3; ++iZ) {
-            pairUbb.push_back(std::make_pair(Form("U_BB_H%dD%dF%dZ%d", iH, iD, iF, iZ), idUbb));
-            ++idUbb;
+            pairUBB.push_back(std::make_pair(Form("U_BB_H%dD%dF%dZ%d", iH, iD, iF, iZ), idUBB));
+            ++idUBB;
           }
         }
       }
     }
 
-    std::vector<std::pair<const std::string, uint>> pairDeadMap;
-    int idDeadMap = 0;
-    for (int iC = 0; iC < 936; ++iC) {
-      pairDeadMap.push_back(std::make_pair(Form("ChipID%d", iC), idDeadMap));
-      ++idDeadMap;
+    for (int iRU = 0; iRU < pairRU.size(); ++iRU) {
+      std::pair<std::string, int> p = pairRU[iRU];
+      mAddressDictRU.emplace(p.second, p.first);
+      mNameDictRU.emplace(p.first, p.second);
+    }
+    for (int iALPIDE = 0; iALPIDE < pairALPIDE.size(); ++iALPIDE) {
+      std::pair<std::string, int> p = pairALPIDE[iALPIDE];
+      mAddressDictALPIDE.emplace(p.second, p.first);
+      mNameDictALPIDE.emplace(p.first, p.second);
+    }
+    for (int iUBB = 0; iUBB < pairUBB.size(); ++iUBB) {
+      std::pair<std::string, int> p = pairUBB[iUBB];
+      mAddressDictUBB.emplace(p.second, p.first);
+      mNameDictUBB.emplace(p.first, p.second);
     }
 
-    for (int iRu = 0; iRu < pairRu.size(); ++iRu) {
-      std::pair<const std::string, int> p = pairRu[iRu];
-      mAddressDictRu.emplace(p.second, p.first);
-      mNameDictRu.emplace(p.first, p.second);
-    }
-    for (int iAlpide = 0; iAlpide < pairAlpide.size(); ++iAlpide) {
-      std::pair<const std::string, int> p = pairAlpide[iAlpide];
-      mAddressDictAlpide.emplace(p.second, p.first);
-      mNameDictAlpide.emplace(p.first, p.second);
-    }
-    for (int iUbb = 0; iUbb < pairUbb.size(); ++iUbb) {
-      std::pair<const std::string, int> p = pairUbb[iUbb];
-      mAddressDictUbb.emplace(p.second, p.first);
-      mNameDictUbb.emplace(p.first, p.second);
-    }
-    for (int iDeadMap = 0; iDeadMap < pairDeadMap.size(); ++iDeadMap) {
-      std::pair<const std::string, int> p = pairDeadMap[iDeadMap];
-      mAddressDictDeadMap.emplace(p.second, p.first);
-      mNameDictDeadMap.emplace(p.first, p.second);
-    }
+    mNameDict.emplace("RU", mNameDictRU);
+    mNameDict.emplace("ALPIDE", mNameDictALPIDE);
+    mNameDict.emplace("UBB", mNameDictUBB);
 
-    mNameDictType.emplace("RU", mNameDictRu);
-    mNameDictType.emplace("ALPIDE", mNameDictAlpide);
-    mNameDictType.emplace("UBB", mNameDictUbb);
-    mNameDictType.emplace("DeadMap", mNameDictDeadMap);
-
-    mAddressDictType.emplace("RU", mAddressDictRu);
-    mAddressDictType.emplace("ALPIDE", mAddressDictAlpide);
-    mAddressDictType.emplace("UBB", mAddressDictUbb);
-    mAddressDictType.emplace("DeadMap", mAddressDictDeadMap);
+    mAddressDict.emplace("RU", mAddressDictRU);
+    mAddressDict.emplace("ALPIDE", mAddressDictALPIDE);
+    mAddressDict.emplace("UBB", mAddressDictUBB);
   }
 
   ClassDefNV(DCSConfigUtils, 1);

--- a/Detectors/ITSMFT/MFT/condition/src/DCSConfigReader.cxx
+++ b/Detectors/ITSMFT/MFT/condition/src/DCSConfigReader.cxx
@@ -10,6 +10,7 @@
 // or submit itself to any jurisdiction.
 
 #include <MFTCondition/DCSConfigReader.h>
+#include "MFTCondition/DCSConfigUtils.h"
 #include "Framework/Logger.h"
 #include "TSystem.h"
 #include <fstream>
@@ -21,15 +22,15 @@ void DCSConfigReader::init(bool ver)
 {
   mVerbose = ver;
   clear();
+  mDCSUtils = new DCSConfigUtils();
+  mDCSUtils->init();
 }
 
 //_______________________________________________________________
 void DCSConfigReader::loadConfig(gsl::span<const char> configBuf)
 {
-
   const char* params = configBuf.data();
   mParams = std::string(params);
-
   parseConfig();
 }
 
@@ -42,73 +43,127 @@ void DCSConfigReader::clear()
 
 void DCSConfigReader::parseConfig()
 {
-
   char delimiter_newline = '\n';
   char delimiter = ',';
 
   std::stringstream stream(mParams);
   std::string row;
 
-  mNumRow = 0;
-  mNumRU = 0;
-  mNumALPIDE = 0;
+  int nrow = 0;
+  int nru = 0;
+  int nalpide = 0;
+  int nubb = 0;
+  int ndeadmap = 0;
 
-  int arrAddRUConf[128];
-  int arrAddALPIDEConf[128];
-
-  int arrValRUConf[128];
-  int arrValALPIDEConf[128];
+  std::string verName = "";
+  std::vector<int> arrRUAdd;
+  std::vector<int> arrRUVal;
+  std::vector<int> arrALPIDEAdd;
+  std::vector<int> arrALPIDEVal;
+  std::vector<std::string> arrUBBName;
+  std::vector<double> arrUBBVal;
+  std::vector<int> arrDeadChip;
 
   while (std::getline(stream, row, delimiter_newline)) {
 
     std::stringstream srow(row);
     std::string col;
 
-    int ncol = 0;
-
     while (std::getline(srow, col, delimiter)) {
+
+      col.erase(std::remove(col.begin(), col.end(), ' '), col.end());
       int val = atoi(col.c_str());
-      if (mNumRow == 0) {
-        arrAddRUConf[ncol] = val;
-        ++mNumRU;
-      } else if (mNumRow == 1) {
-        arrValRUConf[ncol] = val;
-      } else if (mNumRow == 2) {
-        arrAddALPIDEConf[ncol] = val;
-        ++mNumALPIDE;
-      } else if (mNumRow == 3) {
-        arrValALPIDEConf[ncol] = val;
+
+      if (nrow == mDCSUtils->getVersionNameLineInCsv()) {
+        verName = col.c_str();
+      } else if (nrow == mDCSUtils->getRUConfigAddressLineInCsv()) {
+        arrRUAdd.emplace_back(atoi(col.c_str()));
+        ++nru;
+      } else if (nrow == mDCSUtils->getRUConfigValueLineInCsv()) {
+        arrRUVal.emplace_back(atoi(col.c_str()));
+      } else if (nrow == mDCSUtils->getALPIDEConfigAddressLineInCsv()) {
+        arrALPIDEAdd.emplace_back(atoi(col.c_str()));
+        ++nalpide;
+      } else if (nrow == mDCSUtils->getALPIDEConfigValueLineInCsv()) {
+        arrALPIDEVal.emplace_back(atoi(col.c_str()));
+      } else if (nrow == mDCSUtils->getUBBConfigNameLineInCsv()) {
+        col = col.substr(0, 13);
+        arrUBBName.emplace_back(col.c_str());
+        ++nubb;
+      } else if (nrow == mDCSUtils->getUBBConfigValueLineInCsv()) {
+        arrUBBVal.emplace_back(atof(col.c_str()));
+      } else if (nrow == mDCSUtils->getDeadMapLineInCsv()) {
+        arrDeadChip.emplace_back(atoi(col.c_str()));
+        ++ndeadmap;
+      } else {
+        LOG(fatal) << "Not expected parameters are sent from DCS!!!!";
       }
-      ++ncol;
     }
-    ++mNumRow;
+    ++nrow;
   }
+  LOG(info) << "Configuration version / type: " << verName;
 
-  for (int iRUconf = 0; iRUconf < mNumRU; ++iRUconf) {
+  LOG(info) << "Found " << nru << " RU parameters";
+  for (int iRUconf = 0; iRUconf < nru; ++iRUconf) {
     if (mVerbose) {
-      LOG(info) << "(" << arrAddRUConf[iRUconf] << ")   "
-                << " : " << arrValRUConf[iRUconf];
+      LOG(info) << "(" << arrRUAdd[iRUconf] << ")   "
+                << mDCSUtils->getName(arrRUAdd[iRUconf], "RU")
+                << " : " << arrRUVal[iRUconf];
     }
     o2::mft::DCSConfigInfo conf;
     conf.clear();
-    conf.setData(arrValRUConf[iRUconf]);
-    conf.setAdd(arrAddRUConf[iRUconf]);
-    conf.setType(0);
-    conf.setVersion("test");
+    conf.setData(arrRUVal[iRUconf]);
+    conf.setAdd(arrRUAdd[iRUconf]);
+    conf.setType(0); // RU = 0
+    conf.setVersion(verName);
     mDCSConfig.emplace_back(conf);
   }
 
-  for (int iALPIDEconf = 0; iALPIDEconf < mNumALPIDE; ++iALPIDEconf) {
+  LOG(info) << "Found " << nalpide << " ALPIDE parameters";
+  for (int iALPIDEconf = 0; iALPIDEconf < nalpide; ++iALPIDEconf) {
     if (mVerbose) {
-      LOG(info) << "(" << arrAddALPIDEConf[iALPIDEconf] << ")   "
-                << " : " << arrValALPIDEConf[iALPIDEconf];
+      LOG(info) << "(" << arrALPIDEAdd[iALPIDEconf] << ")   "
+                << mDCSUtils->getName(arrALPIDEAdd[iALPIDEconf], "ALPIDE")
+                << " : " << arrALPIDEVal[iALPIDEconf];
     }
     o2::mft::DCSConfigInfo conf;
     conf.clear();
-    conf.setData(arrValALPIDEConf[iALPIDEconf]);
-    conf.setAdd(arrAddALPIDEConf[iALPIDEconf]);
-    conf.setType(1);
-    conf.setVersion("test");
+    conf.setData(arrALPIDEVal[iALPIDEconf]);
+    conf.setAdd(arrALPIDEAdd[iALPIDEconf]);
+    conf.setType(1); // ALPIDE = 1
+    conf.setVersion(verName);
     mDCSConfig.emplace_back(conf);
+  }
+
+  LOG(info) << "Found " << nubb << " UBB parameters";
+  for (int iUBBconf = 0; iUBBconf < nubb; ++iUBBconf) {
+    if (mVerbose) {
+      LOG(info) << "(" << mDCSUtils->getAddress(arrUBBName[iUBBconf], "UBB") << ")   "
+                << arrUBBName[iUBBconf]
+                << " : " << arrUBBVal[iUBBconf];
+    }
+    o2::mft::DCSConfigInfo conf;
+    conf.clear();
+    conf.setData(arrUBBVal[iUBBconf]);
+    conf.setAdd(mDCSUtils->getAddress(arrUBBName[iUBBconf], "UBB"));
+    conf.setType(2); // UBB = 2
+    conf.setVersion(verName);
+    mDCSConfig.emplace_back(conf);
+  }
+
+  LOG(info) << "Found " << ndeadmap << " Dead chips";
+  for (int iDeadMap = 0; iDeadMap < ndeadmap; ++iDeadMap) {
+    if (mVerbose) {
+      LOG(info) << "Chip" << arrDeadChip[iDeadMap] << " is dead.";
+    }
+    o2::mft::DCSConfigInfo conf;
+    conf.clear();
+    conf.setData(1);
+    conf.setAdd(arrDeadChip[iDeadMap]);
+    conf.setType(3); // DeadMap = 3
+    conf.setVersion(verName);
+    mDCSConfig.emplace_back(conf);
+
+    mNoiseMap.maskFullChip(arrDeadChip[iDeadMap], true);
   }
 }

--- a/Detectors/ITSMFT/MFT/condition/testWorkflow/MFTDCSConfigProcessorSpec.h
+++ b/Detectors/ITSMFT/MFT/condition/testWorkflow/MFTDCSConfigProcessorSpec.h
@@ -76,23 +76,47 @@ class MFTDCSConfigProcessor : public o2::framework::Task
 
     using clbUtils = o2::calibration::Utils;
 
-    const auto& payload = mReader.getConfigInfo();
+    ////////////////////////////////////////////////////////////////////////////////////////////////
+    // Preparing the object for DCS Configuration
+    ////////////////////////////////////////////////////////////////////////////////////////////////
 
-    auto clName = o2::utils::MemFileHelper::getClassName(payload);
-    auto flName = o2::ccdb::CcdbApi::generateFileName(clName);
+    const auto& payloadConfigInfo = mReader.getConfigInfo();
+    auto clNameConfigInfo = o2::utils::MemFileHelper::getClassName(payloadConfigInfo);
+    auto flNameConfigInfo = o2::ccdb::CcdbApi::generateFileName(clNameConfigInfo);
 
-    std::map<std::string, std::string> md;
-    md.emplace("created_by", "dpl");
+    std::map<std::string, std::string> mdConfigInfo;
+    mdConfigInfo.emplace("created_by", "dpl");
 
-    o2::ccdb::CcdbObjectInfo info("MFT/Config/Params", clName, flName, md, tf, o2::ccdb::CcdbObjectInfo::INFINITE_TIMESTAMP);
+    o2::ccdb::CcdbObjectInfo infoConfigInfo("MFT/Config/Params", clNameConfigInfo, flNameConfigInfo, mdConfigInfo, tf, o2::ccdb::CcdbObjectInfo::INFINITE_TIMESTAMP);
 
-    auto image = o2::ccdb::CcdbApi::createObjectImage(&payload, &info);
+    auto imageConfigInfo = o2::ccdb::CcdbApi::createObjectImage(&payloadConfigInfo, &infoConfigInfo);
 
-    LOG(info) << "Sending object " << info.getPath() << "/" << info.getFileName() << " of size " << image->size()
-              << " bytes, valid for " << info.getStartValidityTimestamp() << " : " << info.getEndValidityTimestamp();
+    LOG(info) << "Sending object " << infoConfigInfo.getPath() << "/" << infoConfigInfo.getFileName() << " of size " << imageConfigInfo->size()
+              << " bytes, valid for " << infoConfigInfo.getStartValidityTimestamp() << " : " << infoConfigInfo.getEndValidityTimestamp();
 
-    output.snapshot(Output{o2::calibration::Utils::gDataOriginCDBPayload, "DCS_CONFIG_FILE", 0}, *image.get()); // vector<char>
-    output.snapshot(Output{o2::calibration::Utils::gDataOriginCDBWrapper, "DCS_CONFIG_FILE", 0}, info);         // root-serialized
+    output.snapshot(Output{o2::calibration::Utils::gDataOriginCDBPayload, "DCS_CONFIG_FILE", 0}, *imageConfigInfo.get()); // vector<char>
+    output.snapshot(Output{o2::calibration::Utils::gDataOriginCDBWrapper, "DCS_CONFIG_FILE", 0}, infoConfigInfo);         // root-serialized
+
+    ////////////////////////////////////////////////////////////////////////////////////////////////
+    // Preparing the object for Dead Map
+    ////////////////////////////////////////////////////////////////////////////////////////////////
+
+    const auto& payloadDeadMap = mReader.getNoiseMap();
+    auto clNameDeadMap = o2::utils::MemFileHelper::getClassName(payloadDeadMap);
+    auto flNameDeadMap = o2::ccdb::CcdbApi::generateFileName(clNameDeadMap);
+
+    std::map<std::string, std::string> mdDeadMap;
+    mdDeadMap.emplace("created_by", "dpl");
+
+    o2::ccdb::CcdbObjectInfo infoDeadMap("MFT/Calib/DeadMap", clNameDeadMap, flNameDeadMap, mdDeadMap, tf, o2::ccdb::CcdbObjectInfo::INFINITE_TIMESTAMP);
+
+    auto imageDeadMap = o2::ccdb::CcdbApi::createObjectImage(&payloadDeadMap, &infoDeadMap);
+
+    LOG(info) << "Sending object " << infoDeadMap.getPath() << "/" << infoDeadMap.getFileName() << " of size " << imageDeadMap->size()
+              << " bytes, valid for " << infoDeadMap.getStartValidityTimestamp() << " : " << infoDeadMap.getEndValidityTimestamp();
+
+    output.snapshot(Output{o2::calibration::Utils::gDataOriginCDBPayload, "DCS_CONFIG_FILE", 1}, *imageDeadMap.get()); // vector<char>
+    output.snapshot(Output{o2::calibration::Utils::gDataOriginCDBWrapper, "DCS_CONFIG_FILE", 1}, infoDeadMap);         // root-serialized
   }
   //________________________________________________________________
 

